amazon/ion/core.py

```diff
@@ -104,27 +104,27 @@ def is_stream_signal(self):
         return self < 0
 
 
-def scale_to_precision(fractional_seconds, fractional_precision):
+def _scale_to_precision(microseconds, fractional_precision):
     """Scales the ```fractional_seconds``` attribute by -```fractional_precision```.
 
     Args:
-        fractional_seconds (Decimal): The numerical value to scale.
+        microseconds (int): The numerical value to scale.
         fractional_precision (int): The negative of this numerical value to scale ```fractional_seconds``` by.
     Returns:
         Decimal: The scaled value.
     """
-    return Decimal(fractional_seconds).scaleb(-fractional_precision)
+    return Decimal(microseconds).scaleb(-fractional_precision)
```

Comment:
- No, I don't think we should drop `fractional_precision` in favour of number of digits in first parameter. 
Think of the case where in the timestamp we want microseconds of 0.001000. This value will be represented in the int parameter we're passing as as 1000.
If we rely on number of digits, the `Decimal` that will get returned will be `0.1000`, which is incorrect.
- Changed first parameter's name from `microseconds` to `value`.
- Also added logic here to check if `value`'s number of significant digits matches that of `fractional_seconds.` I did this change as I knew previously `reader_text.py` was this kind of check and since it is using this `_scale_to_precision()` method too, I thought it'd be good to put this check here.